packages/flutter_test/lib/src/recording_canvas.dart

```diff
@@ -174,17 +174,18 @@ class TestRecordingPaintingContext extends ClipContext implements PaintingContex
     painter(this, offset);
     canvas.restore();
     final OpacityLayer layer = OpacityLayer();
-    _createdLayers.add(layer);
+    _layersToDispose.add(layer);
```

Comment:
fly-by comment: I am pretty sure that whoever calls `pushOpacity` takes ownership of this layer and is supposed to dispose it.